stumpy/stamp.py

```diff
@@ -126,15 +130,30 @@ def stamp(T_A, T_B, m, ignore_trivial=False):
     T_B.shape[0]-m+1
     """
 
+    # Preprocessing to remove nan and inf values
+    if T_A.ndim != 1:  # pragma: no cover
+        raise ValueError(f"T_A is {T_A.ndim}-dimensional and must be 1-dimensional. ")
+
+    T_A = T_A.copy()
+    T_A[np.isinf(T_A)] = np.nan
     core.check_dtype(T_A)
-    core.check_nan(T_A)
+
+    T_B = T_B.copy()
+    T_B[np.isinf(T_B)] = np.nan
+    if T_B.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T_B is {T_B.ndim}-dimensional and must be 1-dimensional. "
+            "For multidimensional STUMP use `stumpy.mstump` or `stumpy.mstumped`"
+        )
     core.check_dtype(T_B)
-    core.check_nan(T_B)
+
     core.check_window_size(m)
     subseq_T_B = core.rolling_window(T_B, m)
     excl_zone = int(np.ceil(m / 2))
     M_T, Σ_T = core.compute_mean_std(T_A, m)
 
+    T_A[np.isnan(T_A)] = 0
```

Comment:
Please refresh my memory here. So:

1. We take in an array
2. We check to see if there are any `np.inf` values and, if so, then convert them to `np.nan` since `np.inf` cannot be handled properly by `np.mean`/`np.std` but np.nan is fine
3. After the mean and stddev are computed then, according to the [original comment by zpzim](https://github.com/TDAmeritrade/stumpy/issues/115#issuecomment-574500923) that says that:

> You also have to set the time series values which are NaN to zero when actually computing the MP so that your dot product doesn't end up NaN.

we set the `np.nan` values to `0` so that the dot product/distance profile can be computed by `core.mass`/`core.calculate_distance_profile`

4. However, even if a subsequence that has an `np.nan`/`np.inf` within it (and that has been converted temporarily to zero), its distance to any other subsequence will always be np.nan since its mean/stddev is `np.nan`. 

Is this the correct way of thinking about it? Am I missing any logic here that should be called out?